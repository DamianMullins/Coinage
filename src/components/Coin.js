import React from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/Coin.module.scss';
import tick from '../assets/tick.svg';

const Coin = ({ coin, handleOwnedChange }) => (
  <li className={`${styles.coin} ${coin.owned ? styles.coin__owned : ''}`}>
    <label className={styles.label} data-testid="coin-label">
      {coin.owned && <img className={styles.tick} src={tick} alt="" />}

      <img className={styles.image} src={coin.imageUrl} alt="" />

      <input
        className={styles.input}
        type="checkbox"
        checked={coin.owned}
        onChange={handleOwnedChange}
        value={coin.id}
      />
    </label>
  </li>
);

Coin.propTypes = {
  coin: PropTypes.object.isRequired,
  handleOwnedChange: PropTypes.func.isRequired
};

export default Coin;
