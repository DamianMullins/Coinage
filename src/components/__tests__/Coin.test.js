import React from 'react';
import { render } from 'react-testing-library';
import Coin from '../Coin';

it('coin markup is correct', () => {
  // Arrange
  const coin = { id: 2, name: 'Kew Gardens' };
  const component = <Coin coin={coin} handleOwnedChange={() => {}} />;

  // Act
  const { container } = render(component);

  // Assert
  expect(container.firstChild).toMatchSnapshot();
});

it('label text content is correct', () => {
  // Arrange
  const coin = { id: 2 };
  const component = <Coin coin={coin} handleOwnedChange={() => {}} />;

  // Act
  const { getByTestId } = render(component);

  // Assert
  expect(getByTestId('coin-label').textContent).toBe('');
});
