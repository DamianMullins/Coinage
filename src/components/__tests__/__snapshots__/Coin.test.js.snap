// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coin markup is correct 1`] = `
<li
  class="coin "
>
  <label
    class="label"
    data-testid="coin-label"
  >
    <img
      alt=""
      class="image"
    />
    <input
      class="input"
      type="checkbox"
      value="2"
    />
  </label>
</li>
`;
